src/Interpreters/sortBlock.cpp

```diff
@@ -240,33 +310,28 @@ bool isAlreadySorted(const Block & block, const SortDescription & description)
     if (!block)
         return true;
 
-    size_t rows = block.rows();
-
     ColumnsWithSortDescriptions columns_with_sort_desc = getColumnsWithSortDescription(block, description);
+    bool is_collation_required = false;
```

Comment:
Why did we not consider collation before? Is it a bug?

Reply:
It is not a bug. It was mentioned in documentation that collation is not supported for this method.
I added collation support and updated documentation.